src/main/java/org/nazar/service/ParserService.java

```diff
@@ -4,48 +4,71 @@
 import org.jsoup.nodes.Document;
 import org.jsoup.nodes.Element;
 import org.jsoup.select.Elements;
-import org.nazar.util.Constants;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 
 public class ParserService {
-   private String getHtml(String url) throws IOException {
-       return Jsoup.connect(url).get().html();
-   }
-   public List<String> getJavaOffers() throws IOException, InterruptedException {
-       List<String> resultLinks = new ArrayList<>();
-
-       int lastPageNumber = getLastPageNumber(Constants.JOBS_URL);
-
-
-       for (int pageNumber = 1; pageNumber < lastPageNumber; pageNumber++) {
-           String pageUrl = Constants.JOBS_URL + "?page=" + pageNumber;
-           Document document = Jsoup.parse(getHtml(pageUrl));
-           Elements divs = document.select("div.job-list-item");
-           for (Element e : divs) {
-               Element temp = e.select("header").first();
-               if (temp.text().matches(".*\\bJava\\b.*")) {
-                   Element link = temp.select("a.h3.job-list-item__link").first();
-                   resultLinks.add(Constants.BASE_URL + link.attr("href"));
-               }
-           }
-           Thread.sleep(1000);
-       }
+    public static final String BASE_URL = "https://djinni.co";
+    public static final String JOBS_URL = "https://djinni.co/jobs/";
```

Comment:
Why not url like this: https://djinni.co/jobs/?primary_keyword=Java&exp_level=no_exp?